src/main/java/mate/academy/springboot/swagger/resource/ProductResource.java

```diff
@@ -0,0 +1,125 @@
+package mate.academy.springboot.swagger.resource;
+
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.model.dto.mapper.RequestMapper;
+import mate.academy.springboot.swagger.model.dto.mapper.ResponseMapper;
+import mate.academy.springboot.swagger.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.swagger.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.swagger.resource.helper.SortHelper;
+import mate.academy.springboot.swagger.service.ProductService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.util.Streamable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/products")
+public class ProductResource {
```

Comment:
```suggestion
public class ProductСontroller {
```

Reply:
Writing REST applications we operate with resources, don't we? So why we call them controllers and not resources?